### ISSUE TITLE
Officially support Flatpak distribution through Flathub

### ISSUE THREAD

**Issue by Peque** [musescore/MuseScore#20459]
### Your idea

- Even though AppImage is the only officially supported packaging for Linux, this issue aims to gather/measure interest among users in an official Flatpak distribution through Flathub (as al alternative or substitution for the current AppImage).

### Problem to be solved

Easy MuseScore package distribution (installation and updates) through Flathub, already supported in many Linux distros.

### Prior art

There is currently an unofficial version in Flathub: https://github.com/flathub/org.musescore.MuseScore

### Additional context

See previous discussions on this topic:

- https://github.com/musescore/MuseScore/issues/19722#issuecomment-1806410787
- https://github.com/musescore/MuseScore/issues/10622

**Comment by AdrianVovk** [comment]
To the maintainers: Please look at the discussion in https://github.com/musescore/MuseScore/issues/19722! There's some very important arguments as to why AppImage is a poor choice to achieve your stated goals, and how AppImage doesn't actually fulfill most (if _any_) of your stated requirements for a packaging solution. No maintainers have reacted in any way to my comment on that issue, so please take a look.

**Comment by Peque** [comment]
@AdrianVovk Yeah, that discussion was mentioned in the description. Feel free to react with a "thumbs up" to show interest and/or subscribe for future updates. :blush:

**Comment by cbjeukendrup** [comment]
(See also my reply at https://github.com/musescore/MuseScore/issues/19722; I've reopened that issue for the discussion about the FlatPak format, so we can use this issue to track FlatHub)

**Comment by adam248** [comment]
I haven't been able to use MuseScore on my Linux system for about 12 months now due to dependency packages marked by my distro as 'insecure' and 'high risk'. I know that MuseScore claims that they don't use the features that make those dependencies unsafe but that doesn't change the fact that MuseScore has been broken on my system for many months.
The unofficial Flatpak is broken, the AppImage is considered 'unsafe'.
The only solution at this point seems to be either MuseScore dev's switch to offically supporting Flatpak and dropping AppImage or updating their dependencies to up-to-date versions that are not marked 'unsafe'.

Here I am waiting on the sidelines in this silent war of nothingness.... 😮‍💨

**Comment by MarcSabatella** [comment]
Who considers the AppImage “unsafe”?  What specific operation are you performing that reports that and prevents you from simply running it normally?

**Comment by adam248** [comment]
@MarcSabatella don't worry about me and the troubles I'm having. I believe that MuseScore 4.6 will work with the update to Qt6.9+ but it has been a long wait (~12+ months) and I am just very tired in general.

Short answer:
If Flatpak was supported, all the issues I am having just wouldn't even exist. App Image just isn't universal as Flatpaks are.

ps: (side note)
I just checked the downloads for the (unsupported and broken) MuseScore flatpak on flathub and it has been downloaded 20k+ times just since the 4.5.2 update (past 90 days). That is 20,000 people checking to see if the flatpak version is working or not and have similar stories to me (probably). To be honest the Flatpak is 'kinda' working, but not really, it has the same bugs that are stopping my usual way of working with it. Worse case is some of those people are "trying out" MuseScore for the first time and thinking it's not good because Flatpaks are very easily discover-able when people are exploring apps on Linux app stores (most people on Linux don't think to download an executable from random webpages). So there could be many people finding MuseScore via the flatpak, having a poor experience and moving on to some other app. (This would be very sad because MuseScore is amazingly cool and probably the best open-source app in the world!)

Can MuseScore at least start "unofficially and secretly" supporting the flatpak and take ownership over the existing flatpak? It still says: `unverifed` on Flathub's page but says its `by MuseScore BVBA`.
 
link here:
https://flathub.org/apps/org.musescore.MuseScore

Dear whoever is in charge at MuseScore,
Please just have a meeting with the Flathub team.
They are always glad to help an OG open-source project like yourselves.
Just one step at a time. You don't have to aim for 100% straight away.
Thanks

(Oh dear, it seems I rambled a bit on this one and didn't mean to) sorry to anyone who read the whole thing....

**Comment by cbjeukendrup** [comment]
At the moment we cannot prioritise investigating FlatPak, I'm afraid. But if someone creates a PR to this repository to implement FlatPak support, we will be happy to review and accept it.

**Comment by adam248** [comment]
@cbjeukendrup no worries. I know how hard you guys work and understand.
When the time is right, just remember to talk to the Flathub guys, they make the process smooth as butter.

**Comment by cbjeukendrup** [comment]
That's good to know, thanks for the advice!

**Comment by adam248** [comment]
@cbjeukendrup hey, sorry to ping you again. But I forgot to tell you that the code for packaging the current Flatpak on Flathub is an open repo here: https://github.com/flathub/org.musescore.MuseScore

When you are ready to start investigating Flatpak support your build devs can look through the build instructions there.
While there is some patches to try to do workarounds the actual core build instructions are 100 LoC and are in YAML here:
https://github.com/flathub/org.musescore.MuseScore/blob/master/org.musescore.MuseScore.yaml

Maybe if your DevOps guys are feeling curious they could just download the current flatpak and test it out and do some PR's directly to the above repo to see if they can get it working?

The main thing I noticed was the Qt version seemed to be mismatched (at first glance, but not sure).
But with 4.6 version coming you might be able to easily get the flatpak working.
But the *GKDU* might be still an issue.

Anyway, I hope this wasn't too obvious of a tip. I hope you have a great day!

**Comment by krystalcode** [comment]
For me, in 2026, I can't imagine allowing an app to have unrestricted access to the filesystem AND to have internet access - even if its open source. Flatpak would solve these problems but the current Flatpak has some issues with its paths.

I therefore use the distribution's package but restricting it from accessing musescore.org via firewall.
